src/nvim/edit.c

```diff
@@ -5323,6 +5331,9 @@ insertchar (
     do_digraph(buf[i-1]);               /* may be the start of a digraph */
     buf[i] = NUL;
     ins_str(buf);
+    extmark_col_adjust(curbuf, curwin->w_cursor.lnum,
```

Comment:
maybe `mark_col_adjust` should be the entry point around the codebase with a `kExtmarkOnly` flag or something, which then can dispatch to either or both of `extmark_col_adjust` and `oldmark_col_adjust` (or some better name for it). This would make it more convenient to make more stuff use the detailed adjustment (bufhl in its present representation would be a low hanging fruit, though later on bufhl should be integrated fully with extmarks)

Reply:
> maybe mark_col_adjust should be the entry point around the codebase with a kExtmarkOnly flag

Seems like a good idea. Could we make an issue for tracker for things that should be done after this pr is merged?

>  bufhl should be integrated fully with extmarks

yup, again lets track this idea on a list, extmarks should be working properly first :wink:

Reply:
> Could we make an issue for tracker for things that should be done after this pr is merged?

Is this really a such big change that needs deferring tho? we have already changed the signature of `mark_col_adjust` to take a flag for extmark behavior, we can just as well add a flag for "extmark only" at the same time.